assignment/impl/src/java/org/sakaiproject/assignment/impl/persistence/AssignmentRepositoryImpl.java

```diff
@@ -211,9 +212,22 @@ public AssignmentSubmission findSubmissionForUser(String assignmentId, String us
                         canRemove.forEach(s -> deleteSubmission(s.getId()));
                         return submissions.get(0);
                     default:
-                        log.warn("For assignment {} {} submissions found for user: {}, can only remove {} which is not enough to create a unique submission.", assignmentId, submissions.size(), userId, canRemove.size());
+                        log.error("For assignment {} {} submissions found for user: {}, can only remove {} which is not enough to create a unique submission.", assignmentId, submissions.size(), userId, canRemove.size());
+
+                        // delete those we can remove
+                        submissions.removeAll(canRemove);
                         canRemove.forEach(s -> deleteSubmission(s.getId()));
-                        throw new NonUniqueResultException(sizeDiff);
+
+                        // for the rest we don't actually remove the submission but disassociate it from the submitter
+                        // and use the first submission as the one to return to the ui
+                        submissions.sort(Comparator.comparing(AssignmentSubmission::getDateCreated));
```

Comment:
I think we can remove this sort and in submissions add next restriction.
**.addOrder(Order.desc("dateModified")**

_If the assignment is not a group assignment_
After that, iterate submission until getDateSubmitted() != null. 
If we found one, then return this one. If not found return first one.

Adding findSubmissionForUser method code in the conversation.